tests/basic/basic_test.go

```diff
@@ -194,6 +194,51 @@ var _ = Describe("AppTasksDown", func() {
 	})
 })
 
+// This test scales up and down an application and checks if app has actually scaled accordingly
+var _ = Describe("AppScaleUpAndDown", func() {
+	It("has to scale up and scale down the app", func() {
+		var contexts []*scheduler.Context
+		for i := 0; i < Inst().ScaleFactor; i++ {
+			contexts = append(contexts, ScheduleAndValidate(fmt.Sprintf("appscaleupdown-%d", i))...)
+		}
+
+		Step("scale up all deployments/stateful sets ", func() {
+			for _, ctx := range contexts {
+				Step(fmt.Sprintf("updating scale for app: %s", ctx.App.Key), func() {
+					deploymentScaleUpMap, err := Inst().S.GetNewScaleFactorMap(ctx, 1)
+					Expect(err).NotTo(HaveOccurred())
+					err = Inst().S.ScaleApplication(ctx, deploymentScaleUpMap)
+					Expect(err).NotTo(HaveOccurred())
+					err = Inst().S.ScaleStatefulSet(ctx, deploymentScaleUpMap)
```

Comment:
once you merge the two methods above, you won't need two methods here.

Reply:
Fixed.